core-kotlin/src/test/java/com/amplifyframework/kotlin/geo/KotlinGeoFacadeTest.kt

```diff
@@ -0,0 +1,149 @@
+import io.mockk.every
+import io.mockk.mockk
+import junit.framework.Assert.assertSame
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+
+/**
+ * Unit tests for the [KotlinGeoFacade] class.
+ */
+internal class KotlinGeoFacadeTest {
+    private val delegate = mockk<GeoCategoryBehavior>()
+    private val geo = KotlinGeoFacade(delegate)
+
+    @Test
+    fun `gets available maps`() = runBlocking {
+        val maps = listOf(
+            MapStyle("a", "b"),
+            MapStyle("c", "d")
+        )
+        every { delegate.getAvailableMaps(any(), any()) } answers {
+            val callback = firstArg<Consumer<Collection<MapStyle>>>()
+            callback.accept(maps)
+        }
+        val result = geo.getAvailableMaps()
+        assertSame(maps, result)
+    }
+
+    @Test(expected = GeoException::class)
+    fun `throws available map error`(): Unit = runBlocking {
+        val error = GeoException("message", "suggestion")
+        every { delegate.getAvailableMaps(any(), any()) } answers {
+            val callback = secondArg<Consumer<GeoException>>()
+            callback.accept(error)
+        }
+        geo.getAvailableMaps()
+    }
+
+    @Test
+    fun `gets default map`() = runBlocking {
+        val map = MapStyle("name", "style")
+        every { delegate.getDefaultMap(any(), any()) } answers {
+            val callback = firstArg<Consumer<MapStyle>>()
+            callback.accept(map)
+        }
+        val result = geo.getDefaultMap()
+        assertSame(map, result)
+    }
+
+    @Test(expected = GeoException::class)
+    fun `throws default map error`(): Unit = runBlocking {
+        val error = GeoException("message", "suggestion")
+        every { delegate.getDefaultMap(any(), any()) } answers {
+            val callback = secondArg<Consumer<GeoException>>()
+            callback.accept(error)
+        }
+        geo.getDefaultMap()
+    }
+
+    @Test
+    fun `returns map style descriptor`() = runBlocking {
+        val descriptor = MapStyleDescriptor("")
+        every { delegate.getMapStyleDescriptor(any(), any(), any()) } answers {
+            val callback = secondArg<Consumer<MapStyleDescriptor>>()
+            callback.accept(descriptor)
+        }
+        val result = geo.getMapStyleDescriptor()
+        assertSame(descriptor, result)
+    }
+
+    @Test(expected = GeoException::class)
+    fun `throws map style descriptor error`(): Unit = runBlocking {
+        val error = GeoException("message", "suggestion")
+        every { delegate.getMapStyleDescriptor(any(), any(), any()) } answers {
+            val callback = lastArg<Consumer<GeoException>>()
+            callback.accept(error)
+        }
+        geo.getMapStyleDescriptor()
+    }
+
+    @Test
+    fun `returns search by text result`() = runBlocking {
+        val query = "query"
+        val searchResult = GeoSearchResult.withPlaces(emptyList())
+        every { delegate.searchByText(query, any(), any(), any()) } answers {
```

Comment:
For each of the api's with parameters it would be nice to have at least 1 test where any() was replaced with the expected parameters to ensure we are mapping parameters from facade correctly into the callback apis

Reply:
I'll add these for the options. The callbacks have to stay as `any()`.